test/common/stats/thread_local_store_test.cc

```diff
@@ -343,5 +453,135 @@ TEST_F(StatsThreadLocalStoreTest, ShuttingDown) {
   EXPECT_CALL(*this, free(_)).Times(5);
 }
 
+// Histogram tests
+TEST_F(HistogramTest, BasicSingleHistogramMerge) {
+  Histogram& h1 = store_->histogram("h1");
+  EXPECT_EQ("h1", h1.name());
+
+  expectCallAndAccumulate(h1, 0);
+  expectCallAndAccumulate(h1, 43);
+  expectCallAndAccumulate(h1, 41);
+  expectCallAndAccumulate(h1, 415);
+  expectCallAndAccumulate(h1, 2201);
+  expectCallAndAccumulate(h1, 3201);
+  expectCallAndAccumulate(h1, 125);
+  expectCallAndAccumulate(h1, 13);
+
+  EXPECT_EQ(1, validateMerge());
+}
+
+TEST_F(HistogramTest, BasicMultiHistogramMerge) {
+  Histogram& h1 = store_->histogram("h1");
+  Histogram& h2 = store_->histogram("h2");
+  EXPECT_EQ("h1", h1.name());
+  EXPECT_EQ("h2", h2.name());
+
+  expectCallAndAccumulate(h1, 1);
+  expectCallAndAccumulate(h2, 1);
+  expectCallAndAccumulate(h2, 2);
+
+  EXPECT_EQ(2, validateMerge());
```

Comment:
You are not really validating that the merge did something sane because you are not showing the summary. WDYT about adding some data to both histograms so that, when merged, you get a result you expect?

Maybe have one histogram with [1, 1, 0, 0, 0, 0, 0, 0, 1, 1]; and see what the summary for that is.  Make another histogram with [0, 0, 1, 1, 1, 1, 1, 1, 0, 0].  and see what the summary for that is.  EXPECT_EQ on both of those.  Then merge them, and EXPECT_EQ on the summary for that.  Does that make sense?

Reply:
I think the merge logic on our side is validated by the MultiMerge test. Are you trying to validate the libcircl hist_accumulate function? I tested that earlier IIRC and they were not same probably because of the insertion order. They will be close but little different. I will create a test case and send for @postwait's review.

Reply:
There's no test called 'MultiMerge' but there are a number of tests that do merges, which check that the number of merged histograms is as expected.  However, there's none I can see that looked at the values in the merged histogram.  I scanned your message to @postwait and it might be reasonable for there to be some floating point sloppiness, in which case you could do inexact data matching in the test.

But it seems wrong to have no functional testing at all of the merged data.

I'm also fine with changing the data from what I suggested above to something that's easier to reason about in the context of the math behind circlehist.

Reply:
@jmarantz Now I added the test BasicHistogramMergeSummary that validates two set of values 1-50 and 51-100 and validates the merged summary is equal to [1-100].